### ISSUE TITLE
Include tensor shape in its default print formatting

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#63855]
Very often one would print a tensor during debugging instead of checking its shape. It would be useful to have shape information in default tensor print formatting, e.g. at the end after grad_fn or just before grad_fn (so that shape is on the screen if a huge tensor was printed)

**Comment by LoopGlitch26** [comment]
Just for clarification, don't we have torch.shape() and torch.size() for this purpose? @vadimkantorov

**Comment by ozancaglayan** [comment]
IMHO, I think this makes sense as one directly types `x` without thinking too much and then (after thinking a bit :)) the second thing they do is to print `x.shape`. Seen that the `device=` is given as a side information when printing a tensor, I think it's quite useful to have the shape there as well.

**Comment by gchanan** [comment]
I recall a past discussion where we said we'd do this if the format remains repr-able (i.e. that you can copy-paste what is printed and it will give you a tensor).

**Comment by vadimkantorov** [comment]
Even now it's not completely reprable - vast majority of tensors have ellipsis in their print-outs.

Also, maybe a kwarg shape argument could be added to torch.tensor (either a nop-argument or it could check that the passed array indeed matches the passed tensor)...

At least, in practice it would save quite a few keystrokes

**Comment by vadimkantorov** [comment]
@gchanan I found, that shape is already included in tensor display, but only for 0-dim tensors:
```
(Pdb) torch.rand((0, 2))
tensor([], size=(0, 2))
(Pdb) torch.rand((1, 2))
tensor([[0.2159, 0.7472]])
```

**Comment by vadimkantorov** [comment]
So adding this `size=(...)` should be an easy decision for other tensors as well.

Also, is there a naming clash between `size` and `shape`?

**Comment by mpalrocks** [comment]
@vadimkantorov Yes, shape is included as `size=(...)` for the 0-dim tensors [ [Ref ](https://github.com/pytorch/pytorch/blob/008bf6689b7d298e3a788dc5576c9b691c5f25a7/torch/_tensor_str.py#L373) ] and also for the following cases :

1.  meta tensors [ [Ref ](https://github.com/pytorch/pytorch/blob/008bf6689b7d298e3a788dc5576c9b691c5f25a7/torch/_tensor_str.py#L365) ]
2.  sparse tensors [ [Ref ](https://github.com/pytorch/pytorch/blob/008bf6689b7d298e3a788dc5576c9b691c5f25a7/torch/_tensor_str.py#L311) ]
3.  quantized tensors [ [Ref](https://github.com/pytorch/pytorch/blob/008bf6689b7d298e3a788dc5576c9b691c5f25a7/torch/_tensor_str.py#L350) ]

Maybe we can go with `size=(...)` instead of `shape=(...)` as some already includes that.

@gchanan I'm not sure if I got that correctly or not but for example sparse tensors include `size=(...)` in their repr as of now. If we can include default `size=(...)` in repr for all kind of tensors, will that be a problem?

Eg:
![Screenshot from 2021-09-05 15-28-19](https://user-images.githubusercontent.com/37003321/132122846-70c628c4-aa1c-44c4-b75a-4e7af2edb8f9.png)

**Comment by shenmishajing** [comment]
Is this going on? Will we add shape info into repr of tensor?

By the way, how about we add the shape info before the value in repr str? In vscode or pycharm, them will display some chars in head and omit the rest. So, if we put the shape info after value, it will also be difficult to get the shape info.
